cpp/src/copying/concatenate.cu

```diff
@@ -251,15 +247,16 @@ fused_concatenate_kernel(
   while (output_index < output_size) {
 
     // Lookup input index by searching for output index in offsets
-    auto const offset_it = thrust::upper_bound(thrust::seq,
-        input_offsets, input_offsets + input_size, output_index) - 1;
-    size_type const input_index = offset_it - input_offsets;
+    auto const offset_it =
+        thrust::upper_bound(thrust::seq, input_offsets,
+                            input_offsets + num_input_views, output_index) - 1;
+    size_type const partition_index = offset_it - input_offsets;
 
     // Copy input data to output and read bitmask
     bool bit_is_set = false;
-    if (input_index < input_size) {
+    if (partition_index < num_input_views) {
```

Comment:
Nice renaming. Much clearer.

However, I still don't think it should be possible for `partition_index` to ever be larger than `num_input_views`, is it?

Reply:
Yeah, I don't think so either. The only thing I can think this would guard against is if `num_input_views` is > max int32_t, and overflows when converting to `size_type`, but that seems unlikely for a bunch of other reasons too.